File: java/src/main/java/com/dokmatiq/docgen/exception/ValidationException.java
package com.dokmatiq.docgen.exception;

import java.util.Map;

/** 400 – Validation error with field-level details. */
public class ValidationException extends ApiException {
    private final Map<String, String> fieldErrors;
    private final String hint;

    public ValidationException(String message, String responseBody,
                               Map<String, String> fieldErrors, String hint) {
        super(400, message, responseBody);
        this.fieldErrors = fieldErrors;
        this.hint = hint;
    }

    public Map<String, String> fieldErrors() { return fieldErrors; }
    public String hint() { return hint; }
}

File: php/src/Exception/ValidationException.php
<?php

declare(strict_types=1);

namespace Dokmatiq\DocGen\Exception;

/** 400 – Validation error with field-level details. */
class ValidationException extends ApiException
{
    /**
     * @param array<string, string>|null $fieldErrors
     */
    public function __construct(
        string $message,
        ?string $responseBody = null,
        public readonly ?array $fieldErrors = null,
        public readonly ?string $hint = null,
    ) {
        parent::__construct(400, $message, $responseBody);
    }
}
